File: p60057.java
public class p60057 {
    public static int solution(String s) {
        int answer = s.length();
        
        for(int i = 1; i <= s.length() / 2; i++){
            int len = compare(i, s).length();
            answer = Math.min(answer, len);
        }

        return answer;
    }

    public static String compare(int comp, String s){
        String str = "";
        for(int i = 0; i < s.length() + comp; i += comp){
            String s1 = "";

            if(i + comp < s.length()){
                s1 = s.substring(i, i + comp);
            }
            else if(i < s.length()){
                s1 = s.substring(i, s.length());
            }
            else{
                break;
            }

            int cnt = 1;
            int idx = i + comp;

            while(idx + comp <= s.length() && s1.equals(s.substring(idx, idx + comp))){
                cnt++;
                idx += comp;
                i += comp;
            }

            if(cnt > 1){
                str += (Integer.toString(cnt) + s1);
            }
            else{
                str += s1;
            }
        }
        return str;
    }

    public static void main(String[] args){
        System.out.println(solution("aabbaccc"));
        System.out.println(solution("abcabcdede"));
        System.out.println(solution("xababcdcdababcdcd"));
        
    }
}

File: p60057.cpp
#include<bits/stdc++.h>
using namespace std;
string input;
string go(string s, int len){
    string ret = "", prev;
    for(int i = 0; i < s.length(); i += len){
        if(i + len >= s.length()) {
            ret += s.substr(i);
            break;
        }
        else prev = s.substr(i, len);
        int cnt = 1, j = i + len;
        string now = s.substr(j, len);
        while(prev == now && j + len <= s.length()){
            cnt++;
            j += len;
            now = s.substr(j, len);
        }
        if(cnt > 1) ret += (to_string(cnt) + prev);
        else ret += prev;
        i = j - len;
    }
    return ret;
}
int solution(string s) {
    int answer = s.length();
    for(int i = 1; i <= s.length() / 2; i++){
        answer = min(answer, (int)go(s, i).length());
    }
    return answer;
}
int main(){
    cin >> input;
    cout << solution(input) << "\n";
}